client/src/components/friendModal/FriendItem.jsx

```diff
@@ -21,21 +21,10 @@ const FriendItem = ({ friend, friends, onChange, type, name, icon, checked }) =>
   const [clicked, setClicked] = useState(checked);
 
   const handleAddandRemove = () => {
-    // if (friends.includes(id)) {
-    //Remove the friend from the friends list
-    // } else {
-    //Add friend to friends list
-    // console.add(id);
-    console.log('old list', friends);
     setClicked(!clicked);
-    if (checked) {
-      friends = [...friends, friends.filter((e) => e !== friend)];
-    } else {
-      friends = [...friends, friend];
-    }
-    // friends = checked?[...friends.filter((e) => e !== friend)]
-    console.log('new list', friends);
-    onChange(new Set(friends));
+    const newList = [...friends, friend];
+
+    onChange([...new Set(newList)]);
```

Comment:
Doesn't it make more sense to do that on the server side and preventing the same user from being added 2x?

Reply:
Ok I added the logic to check for duplicates of friends in BE
friends.filter(function (value, index, self) {
        return self.indexOf(value) === index;
      });